### PR TITLE
19594 Initial refactoring, styling of other tables

### PR DESCRIPTION
**Pull request by myabc**
:warning: **This PR supersedes #2514, #2758**. Relates to #2766.

https://community.openproject.org/work_packages/19594

Also addresses:
- https://community.openproject.org/work_packages/19099

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://myabc-feature-19594-table-initial-refactoring-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://myabc-feature-19594-table-initial-refactoring-styling.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
This PR is ommiting the part of the ticket that I antocipated the most:

> column headers should be left-aligned, capitalised (similar to work packages table)

No, not the capitalized words (I can live without them). The left-aligned headers. There are tables otu there where I am not able to tell which header belongs to which column. I thought that would finally be fixed :)

File: app/assets/stylesheets/content/_tables.sass [hunk 1]
```diff
@@ -377,3 +369,39 @@ table#time-report
...
+    padding: 0.2em 0.6em 0.2em 0
+    border: 1px solid #e4e4e4
+    border-radius: $button-radius
+    background-color: rgba(255, 255, 255, 0.8)
+    color: darken($content-link-color, 10%)
+
+    &:hover
+      border-color: $button-bg-hover-color
+      background-color: $button-bg-hover-color
+      color: white
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_tables.sass:404]: It looks wrong that the bg color is in a variable and the fg color is not...

What if I want a white bg on hover?

If you say it is too much effort to do it _right_, I'll buy it... but at least I need to ask ^^

File: app/assets/stylesheets/content/_timelines.scss [hunk 2]
```diff
@@ -125,7 +125,7 @@ table.timelines-dates {
     }
   }
   &.timelines-pt td {
-    &.timelines-pt-reorder, &.timelines-pt-allows_association, &.timelines-pt-actions {
+    &.timelines-pt-reorder, &.timelines-pt-allows_association {
```
Comment (NobodysNightmare) [inline app/assets/stylesheets/content/_timelines.scss:128]: At least the part regarding "each on its own line" sounds legit to me.

@myabc What do you say?

**Comment by NobodysNightmare** [comment]
I think `openproject-global_roles` kills the roles table...

edit: Probably [here](https://github.com/finnlabs/openproject-global_roles/blob/release/4.1/app/views/roles/index.html.erb#L42), not sure how that overriding works for views...

**Comment by myabc** [comment]
> This PR is ommiting the part of the ticket that I antocipated the most:
> 
> >  column headers should be left-aligned, capitalised (similar to work packages table)

The work package and PR are currently out-of-sync.

Various requirements were added _after_ I began work on this User Story.

File: app/assets/stylesheets/content/_timelines.scss (hunk 2 above)
Comment (myabc) [inline app/assets/stylesheets/content/_timelines.scss:128]: @NobodysNightmare hopefully all of this code will go away.

File: app/assets/stylesheets/content/_tables.sass (hunk 1 above)
Comment (myabc) [inline app/assets/stylesheets/content/_tables.sass:404]: @NobodysNightmare fair enough.

**Comment by NobodysNightmare** [comment]
> Various requirements were added _after_ I began work on this User Story.

At least you opened the PR after writing the mentioned requirements down, that is where my hope originated...

Not sure what the status of the WP is then, though ^^

**Comment by NobodysNightmare** [comment]
I think it is the costs plugin that is causing trouble on `/projects/chili/settings/members`:

bad-layout

**Comment by myabc** [comment]
@NobodysNightmare 

> Not sure what the status of the WP is then, though ^^

`undefined`.

:wink: will update WP today.

**Comment by NobodysNightmare** [comment]
[app/views/projects/settings/_versions.html.erb](https://github.com/NobodysNightmare/openproject/blob/c1c705d3cfe6343c89426f7e8235dc30af1f17f6/app/views/projects/settings/_versions.html.erb) also gets messed up by a plug-in, I have to find out which one it is...

I am testing on the community configuration, so there are only a few candidates...

edit: `openproject-backlogs` it is! ^^

edit 2: And obviously [app/views/projects/settings/_backlogs_settings.html.erb](https://github.com/opf/openproject-backlogs/blob/release/4.1/app/views/projects/settings/_backlogs_settings.html.erb) is also messed up ^^

**Comment by NobodysNightmare** [comment]
Seems like the messages table of boards was never styled with alternating rows, but I think it would be consequent. After all it is a `list` class table: [app/views/boards/show.html.erb](https://github.com/opf/openproject/blob/5333cda9d95657d95c0ad63f44e920cf79234beb/app/views/boards/show.html.erb)

**Comment by NobodysNightmare** [comment]
Is this PR still on the agenda? I do not want to prioritize it against other bugs you are working on.

But I'd like you to finish this before starting other new features/styling things, because it looked very promising and we need to get stuff done (rather than started ^^)

Disclaimer: I am scrolling through the list of old PRs and therefore asking again here...

**Comment by NobodysNightmare** [comment]
Why did you close this PR? Is there a new one?

**Comment by myabc** [comment]
@NobodysNightmare 

> Why did you close this PR? Is there a new one?

Since this will now target 4.2, I'm planning to rebase and issue a new PR against `dev`. Are you in agreement?

**Pull request closed without merging**